tools/ARIAtools/product.py

```diff
@@ -387,24 +425,36 @@ def __readproduct__(self, fname):
         Read product, determine expected layer names based off of version
         number, and populate corresponding product dictionary accordingly.
         """
+        # enforce NETCDF driver to access metadata
+        basename = os.path.basename(fname)
+        fname = 'NETCDF:"' + fname
+
         # Get standard product version from file
         # version accessed differently between URL vs downloaded product
-        version = str(
-            osgeo.gdal.Open(fname).GetMetadataItem('NC_GLOBAL#version'))
-        if version == 'None':
-            LOGGER.warning(
-                '%s is not a supported file type... skipping', fname)
-            return []
+        # vs NISAR and S1 GUNWs
+        if basename.split('_')[0] == 'NISAR':
```

Comment:
This could be basename.startswith('NISAR_')  which is more simple/clear/lower cognitive load